frontend/src/pages/MessagingQueues/MQDetails/MQDetails.tsx

```diff
@@ -1,65 +1,326 @@
 import './MQDetails.style.scss';
 
 import { Radio } from 'antd';
-import { Dispatch, SetStateAction, useState } from 'react';
+import { QueryParams } from 'constants/query';
+import useUrlQuery from 'hooks/useUrlQuery';
+import { isEmpty } from 'lodash-es';
+import { Dispatch, SetStateAction, useEffect, useMemo, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { AppState } from 'store/reducers';
+import { ErrorResponse, SuccessResponse } from 'types/api';
+import { GlobalReducer } from 'types/reducer/globalTime';
 
 import {
 	ConsumerLagDetailTitle,
-	ConsumerLagDetailType,
+	MessagingQueueServiceDetailType,
+	MessagingQueuesViewType,
+	ProducerLatencyOptions,
+	SelectedTimelineQuery,
 } from '../MessagingQueuesUtils';
 import { ComingSoon } from '../MQCommon/MQCommon';
+import {
+	getConsumerLagDetails,
+	MessagingQueueServicePayload,
+	MessagingQueuesPayloadProps,
+} from './MQTables/getConsumerLagDetails';
+import { getPartitionLatencyDetails } from './MQTables/getPartitionLatencyDetails';
+import { getTopicThroughputDetails } from './MQTables/getTopicThroughputDetails';
 import MessagingQueuesTable from './MQTables/MQTables';
 
+const MQServiceDetailTypePerView = (
+	producerLatencyOption: ProducerLatencyOptions,
+): {
+	[x: string]: MessagingQueueServiceDetailType[];
+} => ({
+	[MessagingQueuesViewType.consumerLag.value]: [
+		MessagingQueueServiceDetailType.ConsumerDetails,
+		MessagingQueueServiceDetailType.ProducerDetails,
+		MessagingQueueServiceDetailType.NetworkLatency,
+		MessagingQueueServiceDetailType.PartitionHostMetrics,
+	],
+	[MessagingQueuesViewType.partitionLatency.value]: [
+		MessagingQueueServiceDetailType.ConsumerDetails,
+		MessagingQueueServiceDetailType.ProducerDetails,
+	],
+	[MessagingQueuesViewType.producerLatency.value]: [
+		producerLatencyOption === ProducerLatencyOptions.Consumers
+			? MessagingQueueServiceDetailType.ConsumerDetails
+			: MessagingQueueServiceDetailType.ProducerDetails,
+	],
+});
+
+interface MessagingQueuesOptionsProps {
+	currentTab: MessagingQueueServiceDetailType;
+	setCurrentTab: Dispatch<SetStateAction<MessagingQueueServiceDetailType>>;
+	selectedView: string;
+	producerLatencyOption: ProducerLatencyOptions;
+}
+
 function MessagingQueuesOptions({
 	currentTab,
 	setCurrentTab,
-}: {
-	currentTab: ConsumerLagDetailType;
-	setCurrentTab: Dispatch<SetStateAction<ConsumerLagDetailType>>;
-}): JSX.Element {
-	const [option, setOption] = useState<ConsumerLagDetailType>(currentTab);
+	selectedView,
+	producerLatencyOption,
+}: MessagingQueuesOptionsProps): JSX.Element {
+	const [option, setOption] = useState<MessagingQueueServiceDetailType>(
+		currentTab,
+	);
+
+	useEffect(() => {
+		setOption(currentTab);
+	}, [currentTab]);
+
+	const handleChange = (value: MessagingQueueServiceDetailType): void => {
+		setOption(value);
+		setCurrentTab(value);
+	};
```

Comment:
this will update `option` and `currentTab` and then the `useEffect` above this line will again update the `option` and we will have one extra re-render here. check if this setOption is needed here ?

Reply:
good catch thanks